Fix failing test: test_merges_history_and_shards

## Summary

The test `tests/test_audit_sharding.py::TestTailMerges::test_merges_history_and_shards` is failing.

## Details

The `ReviewAuditLog.tail()` method is not correctly merging entries from the history file with active shards. It returns only shard entries (2) instead of merged entries (5).

```
AssertionError: assert 2 == 5
```

## Expected Behavior

`tail(n=10)` should return all entries from both the history file and active shards, merged and sorted appropriately.

## Reproduction

```bash
poetry run pytest tests/test_audit_sharding.py::TestTailMerges::test_merges_history_and_shards -v
```

## Priority

Low - this is a pre-existing bug that doesn't affect core workflow functionality.

--- a/tests/test_audit_sharding.py
+++ b/tests/test_audit_sharding.py
@@ -248,7 +248,7 @@ class TestTailMerges:
 
     def test_merges_history_and_shards(self, temp_repo: Path) -> None:
         """Tail should merge entries from history file and active shards."""
-        history_file = temp_repo / "logs" / "governance_history.jsonl"
+        history_file = temp_repo / "logs" / "review_history.jsonl"
 
         # Create history entries
         history_entries = []

